config/initializers/bigquery.rb

```diff
@@ -0,0 +1,12 @@
+require 'google/cloud/bigquery'
+
+BIG_QUERY_API_JSON_KEY = ENV['BIG_QUERY_API_JSON_KEY']
+
+if BIG_QUERY_API_JSON_KEY.present?
+  Google::Cloud::Bigquery.configure do |config|
+    config.project_id  = ENV['BIG_QUERY_PROJECT_ID']
+    config.credentials = JSON.parse(BIG_QUERY_API_JSON_KEY)
```

Comment:
Just curious about the failure mode here. In the initialiser here if the required ENV vars aren't set, then the code continues to try to configure `Google::Cloud::Bigquery` (it doesn't use `ENV.fetch`). In the Sidekiq worker it then requires the ENV vars be set by using `ENV.fetch`.

Are we doing it this way to allow the app to run even if the ENV vars aren't set? This means the Sidekiq workers could die in certain circumstances, would it make more sense to have the initialiser ensure that BigQuery has been setup, and use an actual feature flag to control it?

Reply:
Good point, I think a feature flag to control this would be useful. I suppose we need to know that the configuration is good at application startup and use the same method of configuration in the worker. I'll add as part of this spike PR. I've added a A/C of feature flag to the implementation card.

Reply:
Added feature flag here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4601/commits/9f79d635de040ffecc98bec00456cde054f5f998